2007-12-12  Ulrich Drepper  <drepper@redhat.com>

	[BZ #5465]
	* sysdeps/unix/sysv/linux/x86_64/pthread_cond_timedwait.S [!SHARED]
	(__pthread_cond_timedwait): Don't use VDSO.
	Patch by Michal Januszewski.

===================================================================
RCS file: /cvs/glibc/libc/nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_timedwait.S,v
retrieving revision 1.29
retrieving revision 1.30
diff -u -r1.29 -r1.30
--- libc/nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_timedwait.S	2007/08/20 22:36:19	1.29
+++ libc/nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_timedwait.S	2007/12/12 08:05:33	1.30
@@ -134,12 +134,14 @@
 	/* Only clocks 0 and 1 are allowed so far.  Both are handled in the
 	   kernel.  */
 	leaq	24(%rsp), %rsi
+#ifdef SHARED
 	movq	__vdso_clock_gettime@GOTPCREL(%rip), %rax
 	movq	(%rax), %rax
 	PTR_DEMANGLE (%rax)
 	jz	26f
 	call	*%rax
 	jmp	27f
+#endif
 26:	movl	$__NR_clock_gettime, %eax
 	syscall
 27:
